cmd/snapshot-controller/main.go

```diff
@@ -130,6 +168,11 @@ func main() {
 		*resyncPeriod,
 	)
 
+	if err := ensureCustomResourceDefinitionsExist(snapClient); err != nil {
+		klog.Errorf("Failed to ensure CRDs exist during startup: %+v", err)
```

Comment:
nit: "*Exiting due to failure* to ensure CRDs exist during startup..."

(Please feel free to ignore my nit comments)